src/hotspot/share/opto/regmask.hpp

```diff
@@ -30,9 +30,15 @@
 #include "utilities/count_leading_zeros.hpp"
 #include "utilities/count_trailing_zeros.hpp"
 #include "utilities/globalDefinitions.hpp"
+#include "memory/arena.hpp"
 
 class LRG;
 
+// To avoid unbounded RegMask growth, we need to set a limit on the number of
+// stack slots used by BoxLockNodes. We reach this limit by, e.g., deeply
+// nesting synchronized statements in Java.
+const int BoxLockNode_slot_limit = 200;
```

Comment:
Where does this number come from? I've added arbitrary constants like this, and sometimes it is hard to give a good justification. But at least writing down what was your thinking might help someone else if they come across it later. Do you have a sense how large it should be at least or at most?

Reply:
Thanks, fixed!